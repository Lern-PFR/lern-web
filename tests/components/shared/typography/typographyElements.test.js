import { shallow } from 'enzyme';
import {
	Canon,
	Trafalgar,
	Paragon,
	GreatPrimer,
	DoublePica,
	BodyCopy,
	Pica,
	LongPrimer,
	Brevier,
	Minion,
} from 'components/shared/typography';

describe ('Exported typography elements', () => {
	describe('Canon', () => {
		it('should match previous snapshot', () => {
			const sut = (<Canon>Hello, world!</Canon>);
			const wrapper = shallow(sut);
			
			expect(wrapper).toMatchSnapshot();
		});
	});

	describe('Trafalgar', () => {
		it('should match previous snapshot', () => {
			const sut = (<Trafalgar>Hello, world!</Trafalgar>);
			const wrapper = shallow(sut);
			
			expect(wrapper).toMatchSnapshot();
		});
	});

	describe('Paragon', () => {
		it('should match previous snapshot', () => {
			const sut = (<Paragon>Hello, world!</Paragon>);
			const wrapper = shallow(sut);
			
			expect(wrapper).toMatchSnapshot();
		});
	});

	describe('GreatPrimer', () => {
		it('should match previous snapshot', () => {
			const sut = (<GreatPrimer>Hello, world!</GreatPrimer>);
			const wrapper = shallow(sut);
			
			expect(wrapper).toMatchSnapshot();
		});
	});

	describe('DoublePica', () => {
		it('should match previous snapshot', () => {
			const sut = (<DoublePica>Hello, world!</DoublePica>);
			const wrapper = shallow(sut);
			
			expect(wrapper).toMatchSnapshot();
		});
	});

	describe('BodyCopy', () => {
		it('should match previous snapshot', () => {
			const sut = (<BodyCopy>Hello, world!</BodyCopy>);
			const wrapper = shallow(sut);
			
			expect(wrapper).toMatchSnapshot();
		});
	});

	describe('Pica', () => {
		it('should match previous snapshot', () => {
			const sut = (<Pica>Hello, world!</Pica>);
			const wrapper = shallow(sut);
			
			expect(wrapper).toMatchSnapshot();
		});
	});

	describe('LongPrimer', () => {
		it('should match previous snapshot', () => {
			const sut = (<LongPrimer>Hello, world!</LongPrimer>);
			const wrapper = shallow(sut);
			
			expect(wrapper).toMatchSnapshot();
		});
	});

	describe('Brevier', () => {
		it('should match previous snapshot', () => {
			const sut = (<Brevier>Hello, world!</Brevier>);
			const wrapper = shallow(sut);
			
			expect(wrapper).toMatchSnapshot();
		});
	});

	describe('Minion', () => {
		it('should match previous snapshot', () => {
			const sut = (<Minion>Hello, world!</Minion>);
			const wrapper = shallow(sut);
			
			expect(wrapper).toMatchSnapshot();
		});
	});
});
