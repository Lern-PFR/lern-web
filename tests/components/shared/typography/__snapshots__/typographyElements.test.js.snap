// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exported typography elements BodyCopy should match previous snapshot 1`] = `
<DynamicTextComponent
  fontFamily="IBM Plex Sans, Helvetica Neue, Arial, sans-serif"
  fontSize="1rem"
  fontWeight={400}
  letterSpacing={0}
  lineHeight="1.5em"
  tag="p"
>
  Hello, world!
</DynamicTextComponent>
`;

exports[`Exported typography elements Brevier should match previous snapshot 1`] = `
<DynamicTextComponent
  fontFamily="IBM Plex Sans, Helvetica Neue, Arial, sans-serif"
  fontSize=".778rem"
  fontWeight={400}
  letterSpacing={0.32}
  lineHeight="1.125em"
  tag="p"
>
  Hello, world!
</DynamicTextComponent>
`;

exports[`Exported typography elements Canon should match previous snapshot 1`] = `
<DynamicTextComponent
  fontFamily="IBM Plex Sans, Helvetica Neue, Arial, sans-serif"
  fontSize="3rem"
  fontWeight={300}
  letterSpacing={0}
  lineHeight="4em"
  tag="p"
>
  Hello, world!
</DynamicTextComponent>
`;

exports[`Exported typography elements DoublePica should match previous snapshot 1`] = `
<DynamicTextComponent
  fontFamily="IBM Plex Sans, Helvetica Neue, Arial, sans-serif"
  fontSize="1.444rem"
  fontWeight={400}
  letterSpacing={0}
  lineHeight="2em"
  tag="p"
>
  Hello, world!
</DynamicTextComponent>
`;

exports[`Exported typography elements GreatPrimer should match previous snapshot 1`] = `
<DynamicTextComponent
  fontFamily="IBM Plex Sans, Helvetica Neue, Arial, sans-serif"
  fontSize="1.222rem"
  fontWeight={400}
  letterSpacing={0}
  lineHeight="1.635em"
  tag="p"
>
  Hello, world!
</DynamicTextComponent>
`;

exports[`Exported typography elements LongPrimer should match previous snapshot 1`] = `
<DynamicTextComponent
  fontFamily="IBM Plex Sans, Helvetica Neue, Arial, sans-serif"
  fontSize=".889rem"
  fontWeight={400}
  letterSpacing={0}
  lineHeight="1.375em"
  tag="p"
>
  Hello, world!
</DynamicTextComponent>
`;

exports[`Exported typography elements Minion should match previous snapshot 1`] = `
<DynamicTextComponent
  fontFamily="IBM Plex Sans, Helvetica Neue, Arial, sans-serif"
  fontSize=".778rem"
  fontWeight={400}
  letterSpacing={0.32}
  lineHeight="1.125em"
  tag="p"
  textTransform="uppercase"
>
  Hello, world!
</DynamicTextComponent>
`;

exports[`Exported typography elements Paragon should match previous snapshot 1`] = `
<DynamicTextComponent
  fontFamily="IBM Plex Sans, Helvetica Neue, Arial, sans-serif"
  fontSize="1.778rem"
  fontWeight={400}
  letterSpacing={0}
  lineHeight="2.25em"
  tag="p"
>
  Hello, world!
</DynamicTextComponent>
`;

exports[`Exported typography elements Pica should match previous snapshot 1`] = `
<DynamicTextComponent
  fontFamily="IBM Plex Sans, Helvetica Neue, Arial, sans-serif"
  fontSize="1rem"
  fontWeight={400}
  letterSpacing={0.16}
  lineHeight="1.5em"
  tag="p"
>
  Hello, world!
</DynamicTextComponent>
`;

exports[`Exported typography elements Trafalgar should match previous snapshot 1`] = `
<DynamicTextComponent
  fontFamily="IBM Plex Sans, Helvetica Neue, Arial, sans-serif"
  fontSize="2rem"
  fontWeight={300}
  letterSpacing={0}
  lineHeight="2.5em"
  tag="p"
>
  Hello, world!
</DynamicTextComponent>
`;
